### ISSUE TITLE
introduce puppeteer-core package

### ISSUE THREAD

**Issue by remcohaszing** [puppeteer/puppeteer#1809]
Puppeteer automatically downloads Chrome. This is great, I wish I could use this feature for my tests running in Karma as well, and there are probably more use cases.

I think it would be nice if the Chrome downloader would be extracted from puppeteer, so more tools can depend on it. The export should probably just be the path of the node executable.

E.g.:

```js
const chromePath = require('chrome')

console.log(chromePath) // Something like '/app/node_modules/chrome/bin' or '/app/node_modules/.bin/google-chrome'
```

**Comment by yanivefraim** [comment]
@aslushnikov - I will be happy to help with this task, but I’ll need some help (new repo?)

**Comment by aslushnikov** [comment]
@yanivefraim I'd like to start with a `puppeteer-core` package: puppeteer that doesn't download chromium when installed.

We have a need for the `puppeteer-core` for quite some time, and since pptr itself is pretty slim, it might address this particular usecase.

**Comment by yanivefraim** [comment]
@aslushnikov - great! I would really like to help with this, if possible.  

If it is ok by you (that I will work on this feature) - do you want this package to be in a different repo?

**Comment by aslushnikov** [comment]
> do you want this package to be in a different repo?

@yanivefraim I'd rather have everything in one place for simplicity. The only difference between `puppeteer` and `puppeteer-core` is install script. Ideally, we'd have `package.json` and `package-core.json` and just publish from them, but as far as I understand, npm doesn't let us do this.

> If it is ok by you (that I will work on this feature)

Definitely! I don't think there'll be much code here, but if you explain us how to publish several packages from the same repository, that would be awesome.

**Comment by yanivefraim** [comment]
@aslushnikov - just making sure, did we take into account the option to make Puppeteer repo a monorepo (with two npm packages), using a tool like [lerna](https://github.com/lerna/lerna)?

**Comment by aslushnikov** [comment]
@yanivefraim lerna is quite involved; rolling it in feels like overkill for our small needs.

I'd prefer some tiny ad-hoc solution instead. For example a `utils/publish-puppeteer-core` script that amends `package.json` to change name and remove "install" script, publishes to npm, and then reverts back all the changes.

**Comment by yanivefraim** [comment]
@aslushnikov - 

I guess we're not only ones to have this problem:

https://github.com/travis-ci/travis-ci/issues/5676

I'm still trying to find a way to run two different deployments from one repo/package using Travis' `provider: npm`. Not sure this is possible. 

The alternatives (as far as I can tell) are:
1. Run `puppeteer` deploy as it is today (via Travis' `provider: npm`) + have another script for deploying `puppeteer-core`
2. Run both deployments using a script (Travis' `deploy: script` feature or just adding a 'deploy` script to package.json scripts)

Any thoughts (maybe I miss something here)?

P.S: just making sure - our release versions are being run manually?

**Comment by aslushnikov** [comment]
> I'm still trying to find a way to run two different deployments from one repo/package using Travis' provider: npm. Not sure this is possible.

@yanivefraim I didn't look into this, but it definitely sounds like an interesting idea.

> P.S.: just making sure - our release versions are being run manually

That's right, (almost) all releases are done manually; travis's not involved. The only exception is `puppeteer@next` - a tip-of-tree version of puppeteer that is published automatically by travis.

**Comment by yanivefraim** [comment]
@aslushnikov - yes, so I got it right (: 

If so, I will have to support releasing of `puppeteer-core` as `next`, using Travis (which is what I started investigating) + support manual release of `puppeteer-core` (I guess both will have to use the same util script)

Manual release of `puppeteer-core` should be easy - just run pre and post custom publish script which will update package.json. `Next` release is the tricky part, if we want to use Travis for that.

**Comment by entrptaher** [comment]
Can tools like `lerna` help in such cases?

Just tested `BrowserFetcher` from different file. These toolkits are so powerful and helpful. Extracting them into different packages would help more. 

![image](https://user-images.githubusercontent.com/8284972/40215289-7e2c0f08-5a81-11e8-9495-da4b6c4c0812.png)

**Comment by aslushnikov** [comment]
[`puppeteer-core`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#puppeteer-vs-puppeteer-core) has been published since v1.7.0.
